ocaml/lambda/printlambda.ml

```diff
@@ -170,29 +170,41 @@ let variant_kind print_value_kind ppf ~consts ~non_consts =
       (tag_and_constructor_shape print_value_kind))
     non_consts
 
-let rec value_kind ppf = function
+let value_kind print_value_kind_non_null ppf vk =
+  match vk.nullable with
+  | Non_nullable ->
+    print_value_kind_non_null ppf vk.raw_kind
+  | Nullable ->
+    fprintf ppf "%a or_null" print_value_kind_non_null vk.raw_kind
```

Comment:
Maybe it would be good to have parentheses here, or the list of fields of `Pvariant` will be confusing.

Reply:
I've changed the printing code to generate better results (or at least I hope so). Please re-review when you can.